src/main/java/bio/terra/workspace/service/iam/SamService.java

```diff
@@ -92,11 +102,35 @@ public String getWsmServiceAccountToken() throws IOException {
     return creds.getAccessToken().getTokenValue();
   }
 
+  @FunctionalInterface
+  public interface InterruptedSupplier<R> {
+    R apply() throws InterruptedException;
+  }
+
+  public interface VoidInterruptedSupplier {
+    void apply() throws InterruptedException;
+  }
+
+  public static <T> T rethrowIfSamInterrupted(InterruptedSupplier<T> function, String operation) {
+    try {
+      return function.apply();
+    } catch (InterruptedException e) {
+      throw SamExceptionFactory.create("Interrupted during Sam operation " + operation, e);
+    }
+  }
+
+  public static void rethrowIfSamInterrupted(VoidInterruptedSupplier function, String operation) {
```

Comment:
worth a little comment on how and when to use them. Ok to refer to comments in other places

Reply:
Done.